ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -18,108 +30,227 @@
+          }
+
+          @Override
+          protected void onDetachedFromWindow() {
+            super.onDetachedFromWindow();
+            detach();
+          }
+        };
+    int adLayoutHeight = adSize.getHeightInPixels(context);
+    // If the height is 0 (e.g. for inline adaptive banner requests), use the closest supported size
+    // as the height of the adLayout wrapper.
+    if (adLayoutHeight <= 0) {
+      float density = context.getResources().getDisplayMetrics().density;
+      adLayoutHeight = Math.round(mAdConfig.getAdSize().getHeight() * density);
+    }
+    RelativeLayout.LayoutParams adViewLayoutParams =
+        new RelativeLayout.LayoutParams(adSize.getWidthInPixels(context), adLayoutHeight);
+    adLayout.setLayoutParams(adViewLayoutParams);
+
+    mUniquePubRequestId = config.getRequestUniqueId();
+    if (VungleManager.getInstance().isBannerAdActive(mPlacementId, mUniquePubRequestId)) {
+      // Adapter does not support multiple Banner instances playing for same placement except for
+      // Refresh
+      if (bannerListener != null) {
+        bannerListener.onAdFailedToLoad(adapter, AdRequest.ERROR_CODE_INVALID_REQUEST);
+      }
+      return;
+    }
+
+    mVungleListener = new WeakReference<>(mVungleBannerListener);
+
+    VungleManager.getInstance().storeActiveBannerAd(mPlacementId, VungleBannerAdapter.this);
+    Log.d(TAG, "Requesting banner with ad size: " + mAdConfig.getAdSize());
+    mPendingRequestBanner = true;
+    VungleInitializer.getInstance().initialize(config.getAppId(), context.getApplicationContext(),
+        new VungleInitializer.VungleInitializationListener() {
+          @Override
+          public void onInitializeSuccess() {
+            loadBanner();
+          }
+
+          @Override
+          public void onInitializeError(String errorMessage) {
+            Log.d(TAG, "SDK init failed: " + VungleBannerAdapter.this);
+            VungleManager.getInstance().removeActiveBannerAd(mPlacementId);
+            VungleListener listener = mVungleListener.get();
+            if (mPendingRequestBanner && listener != null) {
+              listener.onAdFailedToLoad(AdRequest.ERROR_CODE_INTERNAL_ERROR);
+            }
+          }
+        });
   }
 
-  // Use weak references to allow VungleInterstitialAdapter to be garbage collected.
-  // Banner view need to be added/removed on adLayout's onAttachedToWindow/onDetachedFromWindow
-  // to break view's parent-child references chain to the leaked VungleBannerAdapter in
-  // VungleManager.
-  void setAdLayout(@NonNull RelativeLayout adLayout) {
-    this.mAdLayout = new WeakReference<>(adLayout);
+  View getBannerView() {
+    Log.d(TAG, "getBannerView # instance: " + hashCode());
+    return adLayout;
   }
 
-  void setVungleListener(@Nullable VungleListener vungleListener) {
-    this.mVungleListener = new WeakReference<>(vungleListener);
+  public void onDestroy() {
+    Log.d(TAG, "onDestroy: " + hashCode());
+    destroy(adLayout);
```

Comment:
this will cause the current display banner ad being destroyed by the previous banner onDestroy().  Because in destroy(View) method `adLayout` always equals to `adView`.

Reply:
Yeah, because the old destroys later...